Dolphin scripts/Entrance Randomizer/lib/entrance_rando.py

```diff
@@ -197,152 +222,280 @@ def highjack_transition_rando():
-                    area.area_id,
-                    area.name,
-                    area.default_entrance,
-                    tuple([
-                        x for x in TRANSITION_INFOS_DICT_RANDO[area.area_id].exits if x != ex
-                    ]),
-                )
-                __connections_left[area.area_id] -= 1
+                delete_exit(area, ex)
+
+
+def initialize_connections_left():
+    for area in transition_infos_dict_rando.values():
+        __connections_left[area.area_id] = len(area.exits)
+
+
+def calculate_mirror(
+    original: Transition,
+    redirect: Transition,
+):
+    mirror_original = Transition(from_=redirect.to, to=redirect.from_)
+    mirror_redirect = Transition(from_=original.to, to=original.from_)
+    return mirror_original, mirror_redirect
+
+
+def create_connection(
+    origin: Transition,
+    redirect: Transition,
+):
+    global total_con_left
+
+    if len(loose_ends) > 0 and __current_hub in {origin.from_, redirect.to}:
+        sacred_pairs.append((origin.from_, redirect.to))
+
+    link_list.append((origin, redirect))
+    _possible_origins_bucket.remove(origin)
+    _possible_redirections_bucket.remove(redirect)
+    if redirect in loose_ends:
+        loose_ends.remove(redirect)
+        sacred_pairs.append((origin.from_, redirect.to))
+
+    mirror_origin, mirror_redirect = calculate_mirror(origin, redirect)
+
+    link_list.append((mirror_origin, mirror_redirect))
+    _possible_origins_bucket.remove(mirror_origin)
+    _possible_redirections_bucket.remove(mirror_redirect)
+    if mirror_redirect in loose_ends:
+        loose_ends.remove(mirror_redirect)
+        sacred_pairs.append((origin.from_, redirect.to))
+
+    __connections_left[origin[0]] -= 1
+    __connections_left[mirror_origin[0]] -= 1
+    total_con_left -= 2
+
+
+def delete_connection(
+    origin: Transition,
+    redirect: Transition,
+):
+    global total_con_left
+
+    link_list.remove((origin, redirect))
+    _possible_origins_bucket.append(origin)
+    _possible_redirections_bucket.append(redirect)
+
+    mirror_origin, mirror_redirect = calculate_mirror(origin, redirect)
+
+    link_list.remove((mirror_origin, mirror_redirect))
+    _possible_origins_bucket.append(mirror_origin)
+    _possible_redirections_bucket.append(mirror_redirect)
+
+    __connections_left[origin[0]] += 1
+    __connections_left[mirror_origin[0]] += 1
+    total_con_left += 2
 
-    # remove exits from ALL_POSSIBLE_TRANSITIONS_RANDO
-    global ALL_POSSIBLE_TRANSITIONS_RANDO
-    for trans in ALL_POSSIBLE_TRANSITIONS:
-        if trans in one_way_exits or trans in disabled_exits:
-            ALL_POSSIBLE_TRANSITIONS_RANDO = [  # pyright: ignore[reportConstantRedefinition]
-                x for x in ALL_POSSIBLE_TRANSITIONS_RANDO if x != trans
-            ]
 
+def choose_random_exit(
+    level: int,
+    bucket_type: int,
+    priority: int,
+):
+    all_exits_available: list[int] = []
```

Comment:
Now assuming Python 3.11 https://github.com/Avasam/ptle-tools/pull/67 these checks can be made as match-case statements so that type-checkers ensure an exhaustive check. Or as simple if-else if you don't expect these enums to ever be more than 2 values

Reply:
Just to be extra clear: I'm not asking you to make a change here.